### PR TITLE
Demonstrate James respond 400 upon unavailable ElasticSearch

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
`FileMailQueueTest.dequeueShouldBeConcurrent » ConditionTimeout Condition with ...`

test this please

File: server/protocols/jmap-draft-integration-testing/cassandra-jmap-draft-integration-testing/src/test/java/org/apache/james/jmap/cassandra/CassandraGetMessageListMethodTest.java [hunk 1]
```diff
@@ -46,4 +55,29 @@ protected GuiceJamesServer createJmapServer() throws IOException {
     protected void await() {
         rule.await();
     }
+
+    @Ignore("Demonstrate James respond 400 upon unavailable ElasticSearch")
+    @Test
+    public void test() throws Exception {
```
Comment (rouazana) [inline server/protocols/jmap-draft-integration-testing/cassandra-jmap-draft-integration-testing/src/test/java/org/apache/james/jmap/cassandra/CassandraGetMessageListMethodTest.java]: name?

File: server/protocols/jmap-draft-integration-testing/cassandra-jmap-draft-integration-testing/src/test/java/org/apache/james/jmap/cassandra/CassandraGetMessageListMethodTest.java [hunk 2]
```diff
@@ -46,4 +55,29 @@ protected GuiceJamesServer createJmapServer() throws IOException {
...
+        rule.getDockerElasticSearchRule().getDockerEs().pause();
+
+        try {
+            given()
+                .header("Authorization", aliceAccessToken.serialize())
+                .body("[[\"getMessageList\", {\"filter\":{\"operator\":\"AND\",\"conditions\":[{\"isFlagged\":\"true\"},{\"isUnread\":\"true\"}]}}, \"#0\"]]")
+            .when()
+                .post("/jmap")
+            .then()
+                .statusCode(500);
```
Comment (mbaechler) [inline server/protocols/jmap-draft-integration-testing/cassandra-jmap-draft-integration-testing/src/test/java/org/apache/james/jmap/cassandra/CassandraGetMessageListMethodTest.java]: or 503?

File: server/protocols/jmap-draft-integration-testing/cassandra-jmap-draft-integration-testing/src/test/java/org/apache/james/jmap/cassandra/CassandraGetMessageListMethodTest.java [hunk 3]
```diff
@@ -46,4 +55,29 @@ protected GuiceJamesServer createJmapServer() throws IOException {
     protected void await() {
         rule.await();
     }
+
+    @Ignore("Demonstrate James respond 400 upon unavailable ElasticSearch")
```
Comment (remk) [inline server/protocols/jmap-draft-integration-testing/cassandra-jmap-draft-integration-testing/src/test/java/org/apache/james/jmap/cassandra/CassandraGetMessageListMethodTest.java]: s/400/500
in commit names too

File: server/protocols/jmap-draft-integration-testing/cassandra-jmap-draft-integration-testing/src/test/java/org/apache/james/jmap/cassandra/CassandraGetMessageListMethodTest.java [hunk 4]
```diff
@@ -58,7 +58,7 @@ protected void await() {
 
     @Ignore("Demonstrate James respond 400 upon unavailable ElasticSearch")
     @Test
-    public void test() throws Exception {
+    public void getMessageListShouldReturn500WhenElasticSearchIsDown() throws Exception {
```
Comment (remk) [inline server/protocols/jmap-draft-integration-testing/cassandra-jmap-draft-integration-testing/src/test/java/org/apache/james/jmap/cassandra/CassandraGetMessageListMethodTest.java]: s/getMessageListShouldReturn500WhenElasticSearchIsDown/getMessageListShouldReturn503WhenElasticSearchIsDown

File: server/protocols/jmap-draft-integration-testing/cassandra-jmap-draft-integration-testing/src/test/java/org/apache/james/jmap/cassandra/CassandraGetMessageListMethodTest.java (hunk 3 above)
Comment (chibenwa) [reply server/protocols/jmap-draft-integration-testing/cassandra-jmap-draft-integration-testing/src/test/java/org/apache/james/jmap/cassandra/CassandraGetMessageListMethodTest.java]: No right now we respond 400, not 503 (which is the expected result)

**Comment by chibenwa** [comment]
`DiscardProtocolTest.testRecord:84 »  Timed out wait for be notified that read ...`

test this please

**Comment by chibenwa** [comment]
Please reword the commit message with the ticket number while merging this: https://issues.apache.org/jira/browse/JAMES-3002

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**